launch/launch/frontend/type_utils.py

```diff
@@ -0,0 +1,254 @@
+
+    :returns: a tuple (type_obj, is_list).
+        is_list is `True` for the supported list types, if not is `False`.
+        type_obj is the object representing that type in python. In the case of list
+        is the type of the items.
+        e.g.:
+            `name = List[int]` -> `(int, True)`
+            `name = bool` -> `(bool, False)`
+            `name = Union[bool, str]` -> `(Union[bool, str], False)`
+            `name = List[Union[bool, str]]` -> `(Union[bool, str], True)`
+            `name = List -> `(None, True)`
+    """
+    is_list = False
+    if data_type is list:
+        is_list = True
+        data_type = None
+    elif check_is_list(data_type):
+        is_list = True
+        data_type = data_type.__args__[0]
+    if data_type is not None and check_valid_scalar_type(data_type) is False:
+        raise ValueError('Unrecognized data type: {}'.format(data_type))
+    return (data_type, is_list)
+
+
+def check_type(value: Any, data_type: Any) -> bool:
+    """
+    Check if `value` is of `type`.
+
+    The allowed types are:
+        - a scalar type: `str`, `int`, `float`, `bool`
+        - a uniform list: `List[str]`, `List[int]`, `List[float]`, `List[bool]`
+        - a non-uniform list.
+            Can specify some of the scalar types for its members, e.g.: List[Union[int, str]].
+            `list` or `List`, means that any of the scalar types are allowed.
+        - an union of the above.
+
+    `types = None` works in the same way as:
+        `Union[int, float, bool, list, str]`
+    """
+    def check_scalar_type(value, data_type):
+        data_type = get_tuple_of_types(data_type)
+        return isinstance(value, data_type)
+    types = __TypesForGuessTuple
+    if data_type is not None:
+        data_type = get_tuple_of_types(data_type)
+    for x in types:
+        type_obj, is_list = extract_type(x)
+        if is_list:
+            if not isinstance(value, list) or not value:
+                continue
+            if type_obj is None:
+                return True
+            if all(check_scalar_type(x, type_obj) for x in value):
+                return True
+        else:
+            if check_scalar_type(value, type_obj):
+                return True
+    return False
+
+
+def coerce_to_bool(x: str) -> bool:
+    """Convert string to bool value."""
+    if x.lower() in ('true', 'yes', 'on', '1', 'false', 'no', 'off', '0'):
+        return x.lower() in ('true', 'yes', 'on', '1')
+    raise ValueError()
+
+
+def coerce_to_str(x: str) -> str:
+    """Strip outer quotes if we have them."""
+    if x.startswith("'") and x.endswith("'"):
+        return x[1:-1]
+    elif x.startswith('"') and x.endswith('"'):
+        return x[1:-1]
+    else:
+        return x
+
+
+def scalar_type_key(data_type: Any) -> int:
+    """Get key. Used for sorting the scalar data_types."""
+    keys = {
+        int: 0,
+        float: 1,
+        bool: 2,
```

Comment:
@ivanpauno meta: I wonder if `bool` should come before `int`

Reply:
No, if not `0` will be converted to bool. If both are possible, it should be an int.
And maybe we should just delete that option. So the rules do not overlap.

Reply:
> If both are possible, it should be an int.

Ok, fair enough, Python coercion will do its job.